linux/system/repo.sls

```diff
@@ -60,13 +98,23 @@ linux_repo_{{ name }}:
   {%- endif %}
   - require:
     - pkg: linux_packages
+  {%- if repo.get('proxy', {}).get('enabled', False) %}
+    - file: /etc/apt/apt.conf.d/95proxies_{{ name }}
+  {%- endif %}
+  {%- if system.proxy.get('pkg', {}).get('enabled', False) %}
+    - file: /etc/apt/apt.conf.d/95proxies
+  {%- endif %}
 
 {%- endif %}
 
 {%- endif %}
 
 {%- if grains.os_family == "RedHat" %}
 
+{%- if repo.get('proxy', {}).get('enabled', False) %}
+# PLACEHOLDER
```

Comment:
describe what's this about in `TODO: blah blah` form to be clear. Or better implement it.